Tasks/Maven/maven2.js

```diff
@@ -133,6 +132,34 @@ function getEndpointDetails(inputFieldName) {
     };
 }
 
+// The endpoint stores the auth details as JSON. Unfortunately the structure of the JSON has changed through time, namely the keys were sometimes Upercase
+// To work around this, we can perform case insenstive checks in the property dictionary of the object. Note that the PowerShell implementation does not suffer from this problem.
+function getAuthParameter(endpoint, paramName) {
+
+    var paramValue = null;
+    var auth = tl.getEndpointAuthorization(endpoint, false);
+
+    if (auth.scheme != "UsernamePassword") {
+        throw new Error("The authorization scheme " + auth.scheme + " is not supported for a SonarQube endpoint. Please use a username and a password.");
+    }
+
+    var parameters = Object.getOwnPropertyNames(auth['parameters']);
+
+    var keyName;
```

Comment:
Yes, but I'd rather keep it like this because it is similar to the implementation here https://github.com/Microsoft/vso-agent/blob/bbabbcab3f96ef0cfdbae5ef8237f9832bef5e9a/src/agent/plugins/release/artifact/jenkinsArtifact.ts - I added a comment about this.